### PR TITLE
Add the current user role check

### PR DESCRIPTION
**Pull request by fisher046**
Check if the user has owner role in the current subscription.

### CODE REVIEW

File: azure/azure-preflight.sh [hunk 1]
```diff
@@ -237,16 +237,29 @@ function store_suggestion()
...
+#### Azure User Role Assignment Checking
+function validate_role_assignment() {
+  user_name=$(echo $1 | jq .user.name | tr -d '"')
+  count=$(az role assignment list --assignee $user_name --include-groups --include-inherited --role Owner -o json | jq length)
+  if [ "$count" = "0" ]; then
+    suggest "Current user $user_name should have Owner role of the subscription $az_subscrb to continue" alert
+  fi
+}
+
+validate_role_assignment "$account"
```
Comment (xugy99) [inline azure/azure-preflight.sh:260]: L256 can `return 1` then

L260 can become

```
validate_role_assignment "$account" || cat $TMP_SUGGESTION || exit 1
```

Comment (fisher046) [reply azure/azure-preflight.sh:260]: After read the whole script, I found each check will not block the whole process. Should we stop here if the user has not enough role?

Comment (xugy99) [reply azure/azure-preflight.sh:260]: I think so. That's how my previous comment was trying to do: if `validate_role_assignment "$account"` return non-zero, then show the suggestion immediately and exit.

Comment (xugy99) [reply azure/azure-preflight.sh:260]: maybe

```
validate_role_assignment "$account" || { cat $TMP_SUGGESTION; exit 1; }
```

Comment (fisher046) [reply azure/azure-preflight.sh]: Modified.

**Review by xugy99 — APPROVED** [review]

**Review by xugy99 — APPROVED** [review]

**Pull request merged**